packages/fuels/tests/harness.rs

```diff
@@ -3721,6 +3683,109 @@ async fn test_rust_result_can_be_encoded() -> Result<(), Box<dyn std::error::Err
+    let log_string =
+        contract_instance._logs_with_type::<SizedAsciiString<4>>(&response.receipts)?;
+    let log_array = contract_instance._logs_with_type::<[u8; 3]>(&response.receipts)?;
+
+    let expected_bits256 = Bits256([
+        239, 134, 175, 169, 105, 108, 240, 220, 99, 133, 226, 196, 7, 166, 225, 89, 161, 16, 60,
+        239, 183, 226, 174, 6, 54, 251, 51, 211, 203, 42, 158, 74,
+    ]);
+
+    assert_eq!(log_u64, vec![64]);
+    assert_eq!(log_bits256, vec![expected_bits256]);
+    assert_eq!(log_string, vec!["Fuel"]);
+    assert_eq!(log_array, vec![[1, 2, 3]]);
+    // ANCHOR_END: produce_logs
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_parse_logs_values() -> Result<(), Error> {
+    setup_contract_test!(
+        contract_instance,
+        wallet,
+        "packages/fuels/tests/test_projects/logged_types"
+    );
+
+    let response = contract_instance.produce_logs_values().call().await?;
+
+    let log_u64 = contract_instance._logs_with_type::<u64>(&response.receipts)?;
+    let log_u32 = contract_instance._logs_with_type::<u32>(&response.receipts)?;
+    let log_u16 = contract_instance._logs_with_type::<u16>(&response.receipts)?;
+    let log_u8 = contract_instance._logs_with_type::<u8>(&response.receipts)?;
+    // try to retrieve non existent log
+    let log_nonexistent = contract_instance._logs_with_type::<bool>(&response.receipts)?;
+
+    assert_eq!(log_u64, vec![64]);
+    assert_eq!(log_u32, vec![32]);
+    assert_eq!(log_u16, vec![16]);
+    assert_eq!(log_u8, vec![8]);
+    assert!(log_nonexistent.is_empty());
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_parse_logs_custom_types() -> Result<(), Error> {
+    setup_contract_test!(
+        contract_instance,
+        wallet,
+        "packages/fuels/tests/test_projects/logged_types"
+    );
+
+    let response = contract_instance.produce_logs_custom_types().call().await?;
+
+    let log_test_struct = contract_instance._logs_with_type::<TestStruct>(&response.receipts)?;
+    let log_test_enum = contract_instance._logs_with_type::<TestEnum>(&response.receipts)?;
+
+    let expected_bits256 = Bits256([
+        239, 134, 175, 169, 105, 108, 240, 220, 99, 133, 226, 196, 7, 166, 225, 89, 161, 16, 60,
+        239, 183, 226, 174, 6, 54, 251, 51, 211, 203, 42, 158, 74,
+    ]);
+    let expected_struct = TestStruct {
+        field_1: true,
+        field_2: expected_bits256,
+        field_3: 64,
+    };
+    let expected_enum = TestEnum::VariantTwo();
+
+    assert_eq!(log_test_struct, vec![expected_struct]);
+    assert_eq!(log_test_enum, vec![expected_enum]);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_print_logs() -> Result<(), Error> {
+    setup_contract_test!(
+        contract_instance,
+        wallet,
+        "packages/fuels/tests/test_projects/logged_types"
+    );
+
+    // ANCHOR: print_logs
+    let response = contract_instance.produce_multiple_logs().call().await?;
+    contract_instance._print_logs(&response.receipts);
```

Comment:
Is the `_print_logs` result deterministic? 

Besides not erroring out, that it actually prints something might be a reasonable assertion.

I'd give them a string and let them do whatever with it rather than print it for them.

Useless if the output is anything other than `stdout`. Besides should it be `stdout` or `stderr`... can't reasonably say as you don't know how the user is going to use it -- as a defacto useful program output or as a diagnostic message.

I'd give them a String :D

Reply:
Yup, a String is more reasonable than printing it out for them.